cms/djangoapps/contentstore/views/course.py

```diff
@@ -776,7 +778,22 @@ def settings_handler(request, course_key_string):
                     # encoder serializes dates, old locations, and instances
                     encoder=CourseSettingsEncoder
                 )
-            else:  # post or put, doesn't matter.
+            else:
+                # post or put, doesn't matter.
+                if settings.FEATURES.get('ENTRANCE_EXAMS', False):
+                    # is entrance exam enabled or not for course.
+                    entrance_exam_enabled = request.json.get('entrance_exam_enabled', '') == 'true'
+                    if entrance_exam_enabled:
+                        # setting to default value from cms configurations.
+                        entrance_exam_minimum_score_pct = float(settings.ENTRANCE_EXAM_MIN_SCORE_PCT)
+                        ee_min_score_pct = request.json.get('entrance_exam_minimum_score_pct', None)
+                        if ee_min_score_pct != '' and ee_min_score_pct is not None:
+                            entrance_exam_minimum_score_pct = float(ee_min_score_pct)
+
+                        create_entrance_exam(request, course_key, entrance_exam_minimum_score_pct)
+                    else:
+                        delete_entrance_exam(request, course_key)
```

Comment:
I'm 99% sure we'll be shifting edx-milestones over to a soft-delete approach, which should avoid the scenario you described -- let me know if additional work would be needed beyond that.